src/libraries/System.Text.Json/src/System/Text/Json/Serialization/ReflectionMemberAccessor.cs

```diff
@@ -73,7 +72,7 @@ internal sealed class ReflectionMemberAccessor : MemberAccessor
             Type type = typeof(T);
 
             Debug.Assert(!type.IsAbstract);
-            Debug.Assert(type.GetConstructors(BindingFlags.Public | BindingFlags.Instance).Contains(constructor));
+            Debug.Assert(Array.IndexOf(type.GetConstructors(BindingFlags.Public | BindingFlags.Instance), constructor) >= 0);
```

Comment:
@stephentoub why don't we have an `Array.Contains` helper for this scenario? Could `Array.Exists` be used here with some predicate, or is that not worth it.

Reply:
> why don't we have an Array.Contains helper for this scenario? 

Do you mean why is there no:
```C#
public static bool Contains<T>(T[] array, T value) => IndexOf(array, value) >= 0;
```
helper?  I think we're talking about the difference between:
```C#
Debug.Assert(Array.IndexOf(type.GetConstructors(BindingFlags.Public | BindingFlags.Instance), constructor) >= 0);
```
and
```C#
Debug.Assert(Array.Contains(type.GetConstructors(BindingFlags.Public | BindingFlags.Instance), constructor));
```
We could add a helper, but... meh :)

> Could Array.Exists be used here with some predicate

Sure:
```C#
Debug.Assert(Array.Exists(type.GetConstructors(BindingFlags.Public | BindingFlags.Instance), c => c == constructor));
```

Reply:
It appears that the `Array.IndexOf` option produces efficient code compared to `Array.Exists`, because latter allocates a closure class for lambda: [sharplab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBLANgHwAEAmARgFgAoQgBgAJDSA6AJRgDNcYwNsIA7ANxUqJOgGEqAbyp05dYBAi46ADVIAKcQIDOGKAFce0AJL92EOpH57DxqAEpZ8lwF4AfHQCCUKAEMATyYzABMYBAB5dg0MAIAHGAho8QcmAHEYDG0bfSMMaB0NACFsfhDSgHMAMVw/Cp0mAAUDYFxsMDp8OhKyypq6hrM9P34wGAc0K11c+wc6d1c6GmFKZzlFZTViLWm7fKgzCymcvegnShc3Tx9/IIBRBGw9QtiEpK1UjKzdvILi0vK/GqtXqTRabQ6XR6gOBA2CORGYwmVjoHhRrkW1lsv0cKwAvkA=) (it's currently using .NET Core 3.1).